src/Riok.Mapperly/Descriptors/SymbolAccessor.cs

```diff
@@ -477,7 +478,25 @@ private bool TryFindPath(ITypeSymbol type, StringMemberPath path, bool ignoreCas
         return symbolMembers.GetValueOrDefault(name);
     }
 
-    public IOperation? GetOperation(SyntaxNode node) => compilationContext.GetSemanticModel(node.SyntaxTree)?.GetOperation(node);
+    public TOperation? GetOperation<TOperation>(SyntaxNode node)
+        where TOperation : class, IOperation => compilationContext.GetSemanticModel(node.SyntaxTree)?.GetOperation(node) as TOperation;
+
+    public ITypeSymbol? GetContainingTypeSymbol(SyntaxNode? node)
+    {
+        var classNode = node;
+        while (classNode is not (null or BaseTypeDeclarationSyntax))
```

Comment:
IMO `classNode is not null and not BaseTypeDeclarationSyntax` would be easier to read here.